### PR TITLE
add customAnnotations in helm chart for serviceaccount

### PR DESCRIPTION
**Pull request by dwerder**
Signed-off-by: Daniel Werdermann <daniel.werdermann@gmail.com>

### Description of your changes

This is a followup for  #2559 . So for example you can use annotations to get your AWS arn there.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

We setup crossplane v1.6.0-rc.0.116.

IRSA trust:
```
{
            "Path": "/",
            "RoleName": "ecr-central-crossplane-prod",
            "RoleId": "AROAX4RJXKBD3V6JMWQ3X",
            "Arn": "arn:aws:iam::123456789112:role/ecr-central-crossplane-prod",
            "CreateDate": "2021-12-28T14:16:28+00:00",
            "AssumeRolePolicyDocument": {
                "Version": "2012-10-17",
                "Statement": [
                    {
                        "Sid": "",
                        "Effect": "Allow",
                        "Principal": {
                            "Federated": "arn:aws:iam::123456789112:oidc-provider/oidc.eks.eu-central-1.amazonaws.com/id/XXXXXXXXXXXXXXXXXXXXXX"
                        },
                        "Action": "sts:AssumeRoleWithWebIdentity",
                        "Condition": {
                            "StringLike": {
                                "oidc.eks.eu-central-1.amazonaws.com/id/XXXXXXXXXXXXXXXX:sub": "system:serviceaccount:crossplane-system:crossplane"
                            }
                        }
                    }
                ]
            },
            "MaxSessionDuration": 3600
        },
```

Permissions for role:
```
{
    "Statement": [
        {
            "Action": [
                "ecr:ListTagsForResource",
                "ecr:ListImages",
                "ecr:GetRepositoryPolicy",
                "ecr:GetLifecyclePolicyPreview",
                "ecr:GetLifecyclePolicy",
                "ecr:GetDownloadUrlForLayer",
                "ecr:GetAuthorizationToken",
                "ecr:DescribeRepositories",
                "ecr:DescribeImages",
                "ecr:DescribeImageScanFindings",
                "ecr:BatchGetImage",
                "ecr:BatchCheckLayerAvailability"
            ],
            "Effect": "Allow",
            "Resource": "*",
            "Sid": "EcrCentralCrossplaneProd"
        }
    ],
    "Version": "2012-10-17"
}
```

Package configuration to be used with AWS ECR
```
apiVersion: pkg.crossplane.io/v1
kind: Configuration
metadata:
  name: test-configuration
spec:
  ignoreCrossplaneConstraints: true
  package: 123456789112.dkr.ecr.eu-central-1.amazonaws.com/crossplane/test-configuration:v0.0.1
```

Result
```
kubectl get configurations
NAME             INSTALLED   HEALTHY   PACKAGE                                                                     AGE
test-configuration   True        True      123456789112.dkr.ecr.eu-central-1.amazonaws.com/crossplane/test-configuration:v0.0.1   10m
```

```
kubectl get sa crossplane -n crossplane-system -o yaml
apiVersion: v1
imagePullSecrets:
- name: dockerhub
kind: ServiceAccount
metadata:
  annotations:
    eks.amazonaws.com/role-arn: arn:aws:iam::123456789112:role/ecr-central-crossplane-prod
...

```

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: cluster/charts/crossplane/values.yaml.tmpl [hunk 1]
```diff
@@ -14,6 +14,10 @@ affinity: {}
 # -- Custom labels to add into metadata
 customLabels: {}
 
+# -- Custom Annotations to add to serviceaccount
+serviceaccount:
```
Comment (hasheddan) [inline cluster/charts/crossplane/values.yaml.tmpl:18]: nitpick: can we use camel case here?
```suggestion
serviceAccount:
```

File: cluster/charts/crossplane/README.md [hunk 2]
```diff
@@ -78,6 +78,7 @@ and their default values.
 | `leaderElection` | Enable leader election for Crossplane Managers pod | `true` |
 | `nodeSelector` | Enable nodeSelector for Crossplane pod | `{}` |
 | `customLabels` | Custom labels to add into metadata | `{}` |
+| `serviceaccount.customAnnotations` | Custom Annotations to add into metadata of sercviceaccount | `{}` |
```
Comment (hasheddan) [inline cluster/charts/crossplane/README.md:81]: Would be good to specify that these are only added to the Crossplane `ServiceAccount`.

**Comment by dwerder** [comment]
Add the suggestions

**Review by hasheddan — APPROVED** [review]
LGTM -- I have tested this out locally and I believe we can include this in `v1.6.0` as it is facilitating usage of a new feature (credentials providers).

**Pull request merged**